sale_subscription_financial_risk/data/account_invoice_check_suspend.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" ?>
+<odoo>
+    <data noupdate="1">
+        <!-- automated action to check a partner record's financial risk status on payment received and suspend/re-activate as needed -->
+        <record id="check_account_service_suspensions" model="base.automation">
+            <field name="name">Suspended Services Check - On Payment</field>
+            <field name="model_id" ref="model_account_invoice"/>
+            <field name="trigger">on_write</field>
+            <field name="active" eval="True"/>
+            <field name="state">code</field>
+            <field name="code">
+record = model.browse(env.context['active_id'])
+if record.run_suspension_check:
```

Comment:
Can we have only one method here which performs the same job in py instead of executing it from XML side

Reply:
I was speaking with @max3903 about this request last week and I believe that I would like to keep these operations on the XML side.  I appreciate the visibility and flexibility this offers operators in their individual instances; i.e. they are able to alter the domain filter in the web UI to handle any unforeseen cases they may be concerned with.